include/drivers/reset.h

```diff
@@ -0,0 +1,325 @@
+
+/**
+ * @brief Static initializer for a @p reset_dt_spec from a DT_DRV_COMPAT
+ * instance's Reset Controller property at an index.
+ *
+ * @param inst DT_DRV_COMPAT instance number
+ * @param idx logical index into "resets"
+ * @return static initializer for a struct reset_dt_spec for the property
+ * @see RESET_DT_SPEC_GET_BY_IDX()
+ */
+#define RESET_DT_SPEC_INST_GET_BY_IDX(inst, idx) \
+	RESET_DT_SPEC_GET_BY_IDX(DT_DRV_INST(inst), idx)
+
+/**
+ * @brief Equivalent to RESET_DT_SPEC_INST_GET_BY_IDX(inst, 0).
+ *
+ * @param inst DT_DRV_COMPAT instance number
+ * @return static initializer for a struct reset_dt_spec for the property
+ * @see RESET_DT_SPEC_INST_GET_BY_IDX()
+ */
+#define RESET_DT_SPEC_INST_GET(inst) \
+	RESET_DT_SPEC_INST_GET_BY_IDX(inst, 0)
+
+/** @cond INTERNAL_HIDDEN */
+
+/**
+ * API template to get the reset status of the device.
+ *
+ * @see reset_status
+ */
+typedef int (*reset_api_status)(const struct device *dev, uint32_t id, uint8_t *status);
+
+/**
+ * API template to put the device in reset state.
+ *
+ * @see reset_assert
+ */
+typedef int (*reset_api_assert)(const struct device *dev, uint32_t id);
+
+/**
+ * API template to take out the device from reset state.
+ *
+ * @see reset_deassert
+ */
+typedef int (*reset_api_deassert)(const struct device *dev, uint32_t id);
+
+/**
+ * API template to reset the device.
+ *
+ * @see reset_toggle
+ */
+typedef int (*reset_api_toggle)(const struct device *dev, uint32_t id);
+
+/**
+ * @brief Reset Controller driver API
+ */
+__subsystem struct reset_driver_api {
+	reset_api_status status;
+	reset_api_assert assert;
+	reset_api_deassert deassert;
+	reset_api_toggle toggle;
+};
+
+/** @endcond */
+
+/**
+ * @brief Get the reset status
+ *
+ * This function returns the reset status of the device.
+ *
+ * @param dev Reset controller device.
+ * @param id Reset line.
+ * @param status Where to write the reset status.
+ *
+ * @retval 0 On success.
+ * @retval -ENOSYS If the functionality is not implemented by the driver.
+ * @retval -errno Other negative errno in case of failure.
+ */
+__syscall int reset_status(const struct device *dev, uint32_t id, uint8_t *status);
+
+static inline int z_impl_reset_status(const struct device *dev, uint32_t id, uint8_t *status)
+{
+	const struct reset_driver_api *api = dev->api;
+
+	if (!api->status) {
+		return -ENOSYS;
+	}
+
+	return api->status(dev, id, status);
+}
+
+/**
+ * @brief Get the reset status from a @p reset_dt_spec.
+ *
+ * This is equivalent to:
+ *
+ *    reset_status(spec->dev, spec->id, status);
+ *
+ * @param spec Reset controller specification from devicetree
+ * @param status Where to write the reset status.
+ *
+ * @return a value from reset_status()
+ */
+static inline int reset_status_dt(const struct reset_dt_spec *spec, uint8_t *status)
+{
+	return reset_status(spec->dev, spec->id, status);
+}
+
+/**
+ * @brief Put the device in reset state
+ *
+ * This function sets/cleares the reset bits of the device,
```

Comment:
Please fix the typo: "clears", not "cleares", here and elsewhere in this patch.

Reply:
Done!